### PR TITLE
defrost `RUST_MIN_STACK=ice rustc hello.rs`

### PR DESCRIPTION
**Pull request by workingjubilee**
I didn't think too hard about testing my previous PR rust-lang/rust#122847 which makes our stack overflow handler assist people in discovering the `RUST_MIN_STACK` variable (which apparently is surprisingly useful for Really Big codebases). After it was merged, some useful comments left in a drive-by review led me to discover I had added an ICE. This reworks the code a bit to explain the rationale, remove the ICE that I introduced, and properly test one of the diagnostics.

### CODE REVIEW

**Comment by workingjubilee** [comment]
Thanks @danielhenrymantilla!

File: compiler/rustc_interface/src/util.rs [hunk 1]
```diff
@@ -51,20 +51,32 @@ pub fn add_configuration(cfg: &mut Cfg, sess: &mut Session, codegen_backend: &dy
...
+            // they might try to "unset" it by running `RUST_MIN_STACK=  rustc code.rs`
+            // this is wrong, but std would nonetheless "do what they mean", so let's do likewise
+            .filter(|s| !s.trim().is_empty())
+            // rustc is a batch program, so error early on inputs which are unlikely to be intended
+            // so no one thinks we parsed them setting `RUST_MIN_STACK="64 megabytes"`
+            // FIXME: we could accept `RUST_MIN_STACK=64MB`, perhaps?
+            .map(|s| {
+                s.trim().parse::<usize>().unwrap_or_else(|_| {
+                    #[allow(rustc::untranslatable_diagnostic)]
+                    early_dcx.early_fatal("`RUST_MIN_STACK` should be unset or a number of bytes")
```
Comment (compiler-errors) [inline compiler/rustc_interface/src/util.rs:70]:
```suggestion
                    early_dcx.early_struct_fatal(format!("`RUST_MIN_STACK` be set to a number of bytes, but it was set to `{}`", s.trim())).note("alternatively, `RUST_MIN_STACK` if it was not meant to be set").emit()
```

Comment (compiler-errors) [reply compiler/rustc_interface/src/util.rs:70]: an idea (modulo formatting) to make the note more descriptive

don't particularly care if it's done with this exact primary/note, but specifically, i'd like for us to explain to the user what *was* set for `RUST_MIN_STACK`, so it's easier for them to (e.g.) search through a bash file or an env file to see where it's set incorrectly.

**Review by compiler-errors — APPROVED** [review]
Do you mind tweaking the message a bit? Left a comment below.

**Review by compiler-errors — APPROVED** [review]

**Comment by fee1-dead** [comment]
r? compiler-errors

**Comment by lqd** [comment]
Then at the very least let's get this to beta as well

**Comment by workingjubilee** [comment]
Beta is currently 1.80, so this is in the beta.

**Comment by lqd** [comment]
Ah sweet!

**Comment by workingjubilee** [comment]
To quote myself over here: https://github.com/rust-lang/rust/issues/126431#issuecomment-2166744641
> I am nominating #125302 for a stable backport. I recommend actually carrying out such a backport only if there are other reasons to do a stable release, at which point it may make sense for it to a catch a ride.

**Comment by apiraino** [comment]
Stable backport approved as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-06-20/near/445862661), with the provision that a dot release is planned - as PR author suggests. 

A backport PR will be authored by the release team at the end of the current development cycle.

@rustbot label +stable-accepted

**Comment by cuviper** [comment]
Clearing stable backport labels since we didn't ship a 1.79 point release, and this is already in current stable 1.80.0.

@rustbot label -stable-nominated -stable-accepted

**Pull request merged**